src/rp2040/hardware_structs/include/hardware/structs/adc.h

```diff
@@ -1,27 +1,86 @@
+// THIS HEADER FILE IS AUTOMATICALLY GENERATED -- DO NOT EDIT
+
 /*
- * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ * Copyright (c) 2021 Raspberry Pi (Trading) Ltd.
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
+
 #ifndef _HARDWARE_STRUCTS_ADC_H
 #define _HARDWARE_STRUCTS_ADC_H
 
 #include "hardware/address_mapped.h"
 #include "hardware/regs/adc.h"
 
+// reference to datasheet: https://datasheets.raspberrypi.org/rp2040/rp2040-datasheet.pdf#tab-registerlist_adc
+// BITMASK : FIELDNAME [BITRANGE] (RESETVALUE): DESCRIPTION
+
 typedef struct {
+    _REG_(ADC_CS_OFFSET)
+    // ADC Control and Status
+    // 0x001f0000 [16-20] : RROBIN (0): Round-robin sampling
+    // 0x00007000 [12-14] : AINSEL (0): Select analog mux input
+    // 0x00000400 [10]    : ERR_STICKY (0): Some past ADC conversion encountered an error
+    // 0x00000200 [9]     : ERR (0): The most recent ADC conversion encountered an error; result is...
```

Comment:
I'll leave it up to @kilograham to decide whether "result is..." is desirable or not.

Reply:
Yeah, I had to do something about the comments. There were duplicate "\n"s, tab separated lines, etc. Splitting by "." and then by a newline was the best compromise that rendered something vaguely useful to the user. Anything longer than 100 chars I truncate.

Reply:
i have changed it to 120